internal/go/skycfg/proto_message.go

```diff
@@ -538,7 +538,21 @@ func typeName(t reflect.Type) string {
 }
 
 func typeError(t reflect.Type, sky starlark.Value) error {
-	return fmt.Errorf("TypeError: value %s (type `%s') can't be assigned to type `%s'.", sky.String(), sky.Type(), typeName(t))
+	if sky.Type() == typeName(t) {
+		return fmt.Errorf(
+			"TypeError: value %s (type `%s') can't be assigned to type `%s'."+
```

Comment:
I'm slightly -1 on this message as is -- it's a lot of text, and it's essentially asking the user to work around a point of poor usability in the library.

Do you think the details could be put into the README, and the error could point people there?

Reply:
Sure, I think that's doable.